Sources/Networking/Logger/URLRequest+LogRepresentation.swift

```diff
@@ -0,0 +1,134 @@
+        
+        let addInfoStringToOutput: (String?) -> Void = {
+            guard let value = $0 else { return }
+            if !output.hasSuffix("\n") { output.append("\n") }
+            output.append(value)
+        }
+        
+        addInfoStringToOutput(getBasicRepresentation())
+
+        if level.contains(.headers) {
+            addInfoStringToOutput(getHeadersRepresentation())
+        }
+        
+        if level.contains(.queryParams) {
+            addInfoStringToOutput(getQueryParamsRepresentation())
+        }
+        
+        if level.contains(.body) {
+            addInfoStringToOutput(getBodyRepresentation())
+        }
+        addSeparatorToOutput()
+        return output.trimmingCharacters(in: .newlines)
+    }
+}
+
+private extension URLRequest {
+    
+    func getBasicRepresentation() -> String {
+        var overviewItems = [(String, String)]()
+
+        if let url = url {
+            overviewItems.append(("URL", url.absoluteString))
+        }
+        
+        if let method = httpMethod {
+            overviewItems.append(("Method", method))
+        }
+        
+        if let requestData = data {
+            let sizeString = ByteCountFormatter.string(fromByteCount: Int64(requestData.count), countStyle: .memory)
+            overviewItems.append(("Request size", sizeString))
+        }
+        
+        return overviewItems
+            .map { "\($0): \($1)" }
+            .joined(separator: "\n")
+    }
+    
+    func getHeadersRepresentation() -> String? {
+        guard let headers = allHTTPHeaderFields, !headers.isEmpty else {
+            return nil
+        }
+        let headersValues = headers
+            .map { "  \"\($0)\": \"\($1)\"" }
+            .joined(separator: ",\n")
+        return ["Headers: [", headersValues, "]"].joined(separator: "\n")
+    }
+    
+    func getQueryParamsRepresentation() -> String? {
+        guard
+            let url = url,
+            let queryItems = URLComponents(url: url, resolvingAgainstBaseURL: false)?.queryItems
+        else { return nil }
+        let queryItemValues = queryItems
+            .map { "  \($0.name) = \($0.value ?? "")" }
+            .joined(separator: "\n")
+        return ["Query items: [", queryItemValues, "]"].joined(separator: "\n")
+    }
+    
+    func getBodyRepresentation() -> String? {
+        let bodyString = data
+            .flatMap { try? JSONSerialization.jsonObject(with: $0, options: []) }
+            .flatMap { try? JSONSerialization.data(withJSONObject: $0, options: .prettyPrinted) }
+            .flatMap { String(data: $0, encoding: .utf8) }
+        guard let unwrappedBody = bodyString, !unwrappedBody.isBlank else { return nil }
+        return ["Body:", unwrappedBody].joined(separator: "\n")
+    }
+}
+
+private extension URLRequest {
+
+    var data: Data? {
+        if let body = httpBody {
+            return body
+        } else if let stream = httpBodyStream {
+            let bufferCount = 4096
```

Comment:
This is just me being curious about 4kB 😄 
Do you have some specific reason why you chose that number, or it seems to you like the fest fit for the json size?

Reply:
Sorry will add the SO link with explanation. Should be power of 2 matching CPU cache block size so 4kB is the best fit in most cases.